agent/cache/cache.go

```diff
@@ -221,7 +221,28 @@ func (c *Cache) Get(t string, r Request) (interface{}, ResultMeta, error) {
 		// once. But be robust against panics.
 		return nil, ResultMeta{}, fmt.Errorf("unknown type in cache: %s", t)
 	}
-	return c.getWithIndex(tEntry, r, r.CacheInfo().MinIndex)
+	return c.getWithIndex(newRequest(tEntry, r))
+}
+
+// request contains the arguments for a Get request. It is used in place of
+// Request so that internal functions can modify Info without having to extract
+// it from the Request each time.
+type request struct {
```

Comment:
I suspect there is a more appropriate name for this struct. `request` is unfortunate as there is already a `Request` interface. Does anyone have suggestions for a better name?

Some ideas I considered, which may still be good options: `getOptions`, `getRequest`.

Reply:
`getOptions` is better than `request` imho. Don't have a better suggestion than that.